exporter/zipkinexporter/factory.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkinexporter
+
+import (
+	"errors"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-service/config/configerror"
+	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/exporter"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "zipkin"
+)
+
+// Factory is the factory for OpenCensus exporter.
+type Factory struct {
+}
+
+// Type gets the type of the Exporter config created by this factory.
+func (f *Factory) Type() string {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for exporter.
+func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
+	return &Config{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+	}
+}
+
+// CreateTraceExporter creates a trace exporter based on this config.
+func (f *Factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+	cfg := config.(*Config)
+
+	if cfg.Endpoint == "" {
+		return nil, nil, errors.New("exporter config requires an Endpoint") // TODO: better error
+	}
+
+	// TODO: (@pjanotti) Move to code like the comment below in a separate PR.
```

Comment:
I do not quite understand the TODO and the commented code that follows it.

Reply:
We are using a piece of code here from openzipikin that does some buffering and is async so even it fails to send the errors are not going to be captured by metrics etc. We just need a simple HTTP post of the data the batching/queueing/retry can be properly provided via processors.

Reply:
Can you remove commented code? I think it is a bad practice.

Reply:
Done, created issue https://github.com/open-telemetry/opentelemetry-service/issues/216